autorag/nodes/promptmaker/fstring.py

```diff
@@ -26,24 +26,25 @@ def _pure(
 		- node_type: prompt_maker
 		  modules:
 		  - module_type: fstring
-			prompt: [Answer this question: {query} \n\n {retrieved_contents},
+			prompt: [Answer this question: {query} \n\n with this message summary {summary} \n\n and these retrieved contents {retrieved_contents},
 			Read the passages carefully and answer this question: {query} \n\n Passages: {retrieved_contents}]
 
 		:param prompt: A prompt string.
 		:param queries: List of query strings.
 		:param retrieved_contents: List of retrieved contents.
+		:param chat_summary: A summary of the chat history.
 		:return: Prompts that are made by f-string.
 		"""
 
 		def fstring_row(
-			_prompt: str, _query: str, _retrieved_contents: List[str]
+			_prompt: str, _query: str, _retrieved_contents: List[str], _chat_summary: str
 		) -> str:
 			contents_str = "\n\n".join(_retrieved_contents)
-			return _prompt.format(query=_query, retrieved_contents=contents_str)
+			return _prompt.format(query=_query, retrieved_contents=contents_str, summary=_chat_summary)
```

Comment:
Also this need to be an optional.

Why don't we get prompt_with_chat as parameter?
If there is no chat_summary input, just use ordinary prompt.
But, when chat_summary is in the input, use prompt_with_chat.

It could be a easy solution without making other chat summary modules in the system.